core/src/test/java/org/kiwix/kiwixmobile/core/history/viewmodel/effects/DeleteHistoryItemsTest.kt

```diff
@@ -0,0 +1,38 @@
+package org.kiwix.kiwixmobile.core.history.viewmodel.effects
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.MutableLiveData
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.kiwix.kiwixmobile.core.dao.HistoryDao
+import org.kiwix.kiwixmobile.core.history.viewmodel.HistoryState
+import org.kiwix.kiwixmobile.core.history.viewmodel.historyItem
+
+internal class DeleteHistoryItemsTest {
+
+  @Test
+  fun `delete with selected items only deletes the selected items`() {
+    val item1 = historyItem(isSelected = true)
+    val item2 = historyItem()
+    val state: MutableLiveData<HistoryState> = mockk()
+    every { state.value } returns HistoryState(listOf(item1, item2), true, "", "")
+    val historyDao: HistoryDao = mockk()
+    val activity: AppCompatActivity = mockk()
+    DeleteHistoryItems(state.value!!, historyDao).invokeWith(activity)
```

Comment:
use your `histroyState()` function here instead of mocking livedata, this class under test doesn't know anything about livedata so the mock is only being called by test code

Reply:
This still needs to be addressed, just delete the liveData, you only use it to create a state object
```
 every { state.value } returns historyState(historyItems = listOf(item1, item2))
DeleteHistoryItems(state.value!!, ...
```
you can just say
```
DeleteHistoryItems(historyState(historyItems = listOf(item1, item2)), ...
```

Reply:
Ah I see what you mean now. Good!